etl/steps/data/garden/artificial_intelligence/2023-06-21/epoch.py

```diff
@@ -0,0 +1,140 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from typing import cast
+
+import numpy as np
+import pandas as pd
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("epoch.start")
+
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = cast(Dataset, paths.load_dependency("epoch"))
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["epoch"]
+    df = pd.DataFrame(tb)
+    #
+    # Process data.
+    # Clean up researcher affiliation
+    category_mapping = {
+        "Industry - Academia Collaboration": "Collaboration",
+        "Industry - Academia collaboration": "Collaboration",
+        "Industry - Academia Collaboration (Academia leaning)": "Collaboration, majority academia",
+        "Industry - Academia Collaboration (Academia Leaning)": "Collaboration, majority academia",
+        "Industry - Academia Collaboration (Industry Leaning)": "Collaboration, majority industry",
+        "Industry - Academia Collaboration (Industry leaning)": "Collaboration, majority industry",
+        "Research Collective": "Research collective",
+    }
+    df["organization_categorization"] = df["organization_categorization"].replace(category_mapping)
+
+    # Convert FLOP to petaFLOP
+    df["training_computation_petaflop"] = df["training_compute__flop"] / 10e14
+
+    # apply the function to 'publication_date'
+    df["publication_date"] = df["publication_date"].apply(convert_date)
+
+    # then, calculate 'days_since_1949'
+    df["days_since_1949"] = (df["publication_date"] - pd.to_datetime("1949-01-01")).dt.days
+
+    df.dropna(subset=["days_since_1949", "system"], inplace=True)
+    df = df.reset_index(drop=True)
+    # df.drop("publication_date", axis=1, inplace=True)
+    df["days_since_1949"] = df["days_since_1949"].astype(int)
+
+    df = clean_non_unique_year_model(df)
+
+    df.replace({"VIsion": "Vision"}, inplace=True)
```

Comment:
This seems to only apply to column `domain`? If so, I'd add a comment and apply the mapping only to this column:

```python
# There is a typo in the domain name "VIsion"
df["domain"] = df["domain"].replace({"VIsion": "Vision"})
```